src/tests/JIT/Regression/JitBlue/Runtime_62692/Runtime_62692.cs

```diff
@@ -39,8 +40,16 @@ public static int TestEntryPoint()
             AssertEqual(Problem2(1111, 0xFFFF_FFFF_0000_0001), 3414328792);
             AssertEqual(Problem3(1, 0xFFFF_0001), 0);
             AssertEqual(Problem4(1111, 0xFFFF_FFFF_0000_0001), 3414328792);
-            AssertEqual(Problem5(1111, double.MaxValue), 3307008522);
-            AssertEqual(Problem6(1111, float.MaxValue), 3307008522);
+            if (RuntimeInformation.ProcessArchitecture == Architecture.X64 || RuntimeInformation.ProcessArchitecture == Architecture.X86)
```

Comment:
Why is this needed? I would expect this PR to eliminate differences between platforms, not introduce them.

Reply:
There is still a difference in implementations because of ```Arm``` using ```Backward_Compatible``` in some cases.
https://github.com/khushal1996/runtime/blob/f3e4bf5556f730dec35c8951aa97ba9480f5413b/src/tests/JIT/Directed/Convert/out_of_range_fp_to_int_conversions.cs#L176C1-L198C14

```
            switch (t)
            {
                case FPtoIntegerConversionType.CONVERT_BACKWARD_COMPATIBLE:
                    return (Double.IsNaN(x) || (x < long.MinValue) || (x >= ullong_max_plus_1)) ? unchecked((ulong)long.MinValue): (x < 0) ? (ulong)(long)x: (ulong)x;


                case FPtoIntegerConversionType.CONVERT_SENTINEL:
                    return (Double.IsNaN(x) || (x < 0) || (x >= ullong_max_plus_1)) ? ulong.MaxValue : (ulong)x;


                case FPtoIntegerConversionType.CONVERT_SATURATING:
                    return (Double.IsNaN(x) || (x < 0)) ? 0 : (x >= ullong_max_plus_1) ? ulong.MaxValue : (ulong)x;


                case FPtoIntegerConversionType.CONVERT_MANAGED_BACKWARD_COMPATIBLE_ARM32:
                    {
                        if (x < two63)
                        {
                            return (ulong)ConvertDoubleToInt64(x, FPtoIntegerConversionType.CONVERT_MANAGED_BACKWARD_COMPATIBLE_ARM32);
                        }
                        else
                        {
                            return (ulong)ConvertDoubleToInt64(x - two63, FPtoIntegerConversionType.CONVERT_MANAGED_BACKWARD_COMPATIBLE_ARM32) + (0x8000000000000000);
                        }
                    }
            }
```

https://github.com/khushal1996/runtime/blob/f3e4bf5556f730dec35c8951aa97ba9480f5413b/src/tests/JIT/Directed/Convert/out_of_range_fp_to_int_conversions.cs#L356C1-L369C14
```
        public static int TestEntryPoint()
        {
            switch (RuntimeInformation.ProcessArchitecture)
            {
                case Architecture.Arm:
                    Program.ManagedConversionRule = FPtoIntegerConversionType.CONVERT_MANAGED_BACKWARD_COMPATIBLE_ARM32;
                    break;


                case Architecture.X86:
                case Architecture.X64:
                case Architecture.Arm64:
                    Program.ManagedConversionRule = FPtoIntegerConversionType.CONVERT_SATURATING;
                    break;
            }
```

Arm is not saturating in all cases. Hence, there are these separate implementations. Let me know if I am wrong. Although the purpose of this PR is to saturate ```x86/x64``` without changing anything else. Hence, things were kept as they were for arm.

Reply:
I do not see the point in adding extra checks here or in https://github.com/dotnet/runtime/pull/97529#discussion_r1534709378 that somebody will need to go a submit a follow up PR to delete to get the behavior unified.

Reply:
I understand that there will still be a followup for changing behavior on arm32. We should discuss this since this was not in the scope of the PR. We were targeting Intel IA for these changes.

@tannergooding @DeepakRajendrakumaran 
So what is happening here that the helpers now have following structure --
```
#ifdef TARGET_XARCH
...
#else
...
#endif
```

This is mainly because ```arm32``` does not saturate as of now. Do we want to make the change in this PR since this was targeted at Intel IA. I see that ```arm32``` has a different behavior and it does not saturate.

Reply:
If there is extra code to be written to get the arm32 behavior unified, it is absolutely fine to do that in a follow up PR.

My objection is around artificially complicating the code to keep arm32 intact.

Reply:
Here is a commit that would address my concern: https://github.com/jkotas/runtime/commit/d171afd7ec1301a2568f58a72737ee090e3a3a20 (feel free to cherry pick it).

It deletes unnecessary/broken code and fixes `JIT_Dbl2ULngOvf` helper to the saturating behavior. We seem to be missing the test coverage for this helper give that it did not show up as a test failure on x86/x64.

Reply:
(update a link to the commit)

Reply:
> fixes JIT_Dbl2ULngOvf helper to the saturating behavior

It is possible that there is no actual observable bug, but it is hard to tell by looking at the code.

Reply:
> If there is extra code to be written to get the arm32 behavior unified, it is absolutely fine to do that in a follow up PR.
> 
> My objection is around artificially complicating the code to keep arm32 intact.

Sure. I will discuss this with the team and see if whether we should cover this in the current PR or not.

Reply:
> Here is a commit that would address my concern: [jkotas@fe60ff2](https://github.com/jkotas/runtime/commit/fe60ff258de6ea91ef0943b18f767a4f9a956777) (feel free to cherry pick it).
> 
> It deletes unnecessary/broken code and fixes `JIT_Dbl2ULngOvf` helper to the saturating behavior. We seem to be missing the test coverage for this helper give that it did not show up as a test failure on x86/x64.

Thanks. I will cherry pick these changes. The changes would fail one of the tests. I will see if we can make it right. thanks  for the sample commit.

Reply:
>  I did not find a case where x86/x64 would go to JIT_Dbl2ULngOvf

This helper is definitely used on x86/x64. As I have said it is possible that we are missing the test coverage for this helper. Run the following with your change :
```
using System;

for (;;) Console.WriteLine(checked((ulong)Random.Shared.NextDouble()).ToString());
```
You should see the `JIT_Dbl2ULngOvf` helper executed.

Reply:
@jkotas thanks for this review. I have removed ```FastDbl2Lng``` and also added overflow handling for ``x86/x64```. Hope this helps.